Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -1722,11 +1709,14 @@ namespace Babylon
                     xr::Anchor& nativeAnchor = xrAnchor->GetNativeAnchor();
 
                     // Update the anchor, and validate it is still a valid anchor if not the remove from the collection.
-                    m_frame->UpdateAnchor(nativeAnchor);
+                    if (nativeAnchor.IsValid)
+                    {
+                        m_frame->UpdateAnchor(nativeAnchor);
+                    }
 
                     if (!nativeAnchor.IsValid)
```

Comment:
We actually need to redo this check because it can be marked invalid by the system as well as by an explicit user delete if we lose tracking on the anchor.  #340 is a proposal from Justin to basically eliminate this code from UpdateAnchors, which would let us get rid of this double check.

Reply:
Added a comment to clarify why we have two checks